src/registrar/admin.py

```diff
@@ -1012,6 +1028,71 @@ def save_model(self, request, obj, form, change):
                 "This action is not permitted for applications with a restricted creator.",
             )
 
+            return None
+        
+        if change:
```

Comment:
reminder: if you find a bug make a ticket for it so we can be sure it's linked as part of the resolve. Also would suggest with a PR this big a bug fix would best go into another PR instead. No changes needed though on this one since I see it already has two approvals. Just more of an fyi